agent/app/agent.go

```diff
@@ -454,11 +460,19 @@ func (agent *ecsAgent) setClusterInConfig(previousCluster string) error {
 
 // getEC2InstanceID gets the EC2 instance ID from the metadata service
 func (agent *ecsAgent) getEC2InstanceID() string {
-	instanceID, err := agent.ec2MetadataClient.InstanceID()
+	var instanceID string
+	var err error
+	backoff := retry.NewExponentialBackoff(getInstanceIdBackoffMin, getInstanceIdBackoffMax, getInstanceIdBackoffJitter, getInstanceIdBackoffMultiple)
+	for i := 0; i < 3; i++ {
+		instanceID, err = agent.ec2MetadataClient.InstanceID()
```

Comment:
could you check if the error message equals "blackholed" https://github.com/aws/amazon-ecs-agent/blob/master/agent/ec2/blackhole_ec2_metadata_client.go#L53 and don't retry in that case? we use the blackholed client when running on-prem and we don't want to retry in that case

Reply:
Thanks for calling this out! it's fixed now.